sysgen/sysgen.go

```diff
@@ -501,6 +501,12 @@ func generateArg(
 			failf("wrong number of arguments for %v arg %v, want %v, got %v", typ, name, want, len(a))
 		}
 		fmt.Fprintf(out, "PtrType{%v, Dir: %v, Type: StrConstType{%v, Val: \"%v\"}}", common(), fmtDir("in"), common(), a[0]+"\\x00")
+	case "bufconst":
+		canBeArg = true
```

Comment:
Remove this. canBeArg means that the type can be a direct argument of a syscall, e.g.:
```
write(..., x bufconst["foo"], ...)
```
It does not make sense for bufconst, as syscall arguments must be uintptr-sized, while bufconst can be of arbitrary size. There is no way to pass it to a syscall.